src/arrayMethodJoin.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = typeof separator === 'undefined' ? ',' : separator;
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (element === null || typeof element === 'undefined') {
+        if (i === this.length - 1) {
```

Comment:
Remove unnecessary check for the last element, as it is handled by the loop condition.